src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/restartDuringPodLaunch.groovy

```diff
@@ -2,6 +2,15 @@ podTemplate(yaml: '''
 apiVersion: v1
 kind: Pod
 spec:
+  containers:
+  - name: jnlp
+    resources:
+      requests:
+        cpu: 100m
+        memory: 256Mi
+      limits:
+        cpu: 100m
+        memory: 256Mi
```

Comment:
If I remove this, then the test passes O_o

Reply:
The CPU limit would be throttling the pod and changing timing conditions. Maybe that is why.

Reply:
At least _Waiting for agent to connect_ could be reworded to emphasize that the controller might be waiting for the agent to be fully initialized and ready to use.